### DISCUSSION TITLE
How to detect a language with SpaCy. Is that at all possible without any other module?

### DISCUSSION THREAD

**Discussion by joseberlines** [explosion/spaCy#12199] [Help: Other Questions] [answered]
When looking online (and using chatgpt) I get that it was possible to detect a language of a text by:

```
import spacy
text = "This is an example text."
language = spacy.util.detect_language(text)
print(language)
```

I am working with spacy 3.5.0.

and this way of detecting a language of a text is not possible anymore.

Is there any other way?

I am redirected to use langdetect as in:

```
!pip install langdetect
from langdetect import detect
text = "This is an example text."
language = detect(text)
print(language)
```

but this requieres an extra library that is a big inconvinient for me.
Is it at all possible to detect the laguage of a text using only spacy?

**Comment by polm** [comment, marked as answer]
Sorry, language detection is not a core feature of spaCy and never has been, and there is no `spacy.util.detect_language` function. You need to be careful with the output of ChatGPT, as it will happily make up things that do not exist.
